src/middleware/nft.middleware.ts

```diff
@@ -1,64 +1,65 @@
 
-    // Obtén la fecha actual en UTC
-    const DateUTC = new Date();
+  use(req: Request, res: Response, next: NextFunction): void {
+    try {
+      const data: any = {};
 
-    // Ajusta la zona horaria a UTC-5
-    const DateUTCMinus5 = new Date(DateUTC.getTime() - (5 * 60 * 60 * 1000));
+      data.ip = req.headers['x-real-ip'] || req.ip;
+      data.call_date = new Date(new Date().getTime() - (5 * 60 * 60 * 1000)).toISOString();
+      data.endpoint = `${req.method} ${req.url}`;
 
-    // Formatea la fecha en formato ISO
-    const time = DateUTCMinus5.toISOString();
+      if (req.body && Object.keys(req.body).length) {
+        data.request = JSON.stringify(req.body);
+      }
 
-    console.log("=> Datos del Middleware");
-    console.log(`Request: ${req.method} ${req.url}`);
-    console.log(`IP: ${ip}, Time: ${time}`);
+      const authorizationToken = req.headers['authorization']?.substring(7);
+      data.accessToken = authorizationToken;
 
-    // Imprimir el body del request si es un JSON
-    if (req.body && Object.keys(req.body).length) {
-      console.log("Request JSON:");
-      console.log(req.body);
-    }
+      if (authorizationToken) {
+        const jwt = require('jsonwebtoken');
+        const decodedToken = jwt.decode(authorizationToken);
+        if (decodedToken && decodedToken.id) {
+          data.projectId = decodedToken.id;
+        } else {
+          this.logger.error('Invalid token or missing ID');
+        }
+      } else {
+        this.logger.error('Authorization token missing');
+      }
 
-    const authorizationToken = req.headers['authorization'].substring(7); 
-    console.log('Authorization Token:', authorizationToken);
+      const originalSend = res.send;
+      res.send = function (body): any {
+        if (body !== undefined) {
+          data.response = body;
+        }
 
-    const jwt = require('jsonwebtoken');
-    const token = authorizationToken
-    const id = jwt.decode(token).id;
-    console.log('ID:', id);
+        data.statusCode = res.statusCode;
 
-    // const originalSend = res.send;
-    res.send = function(body): any {
-      if (body !== undefined) {
-        // Si hay un cuerpo de respuesta, imprímelo
-        console.log("Response JSON:");
-        console.log(body);
-      }
-    
-      // Imprimir el código de estado de la respuesta
-      console.log("Status Code:", res.statusCode);
-    
-      // No hagas nada más, solo imprime en consola
-    };
+        const setFinishTime = new Date();
+        const responseTime = setFinishTime.getTime() - new Date(data.call_date).getTime();
+        data.duration = `${responseTime}ms`;
 
-    res.on('finish', () => {
-      // Obtener la fecha actual cuando se envía la respuesta
-      const setFinishTime = new Date();
+        async function sendData() {
+          console.log(" => body para la base de datos", data);
+          await axios.post('http://localhost:8080/api/requests', data);
```

Comment:
@TincoNomad esto está bien, sin embargo dado que el registro del request está siendo realizado dentro del mismo proyecto sería suficiente con usar el service que creaste y ya no sería usar el controller. El controller solo nos sirve para realizar pruebas desde postman.
En conclusión sugiero cambiar esa línea por esta:
`await this.requestService.create(apiTrackingInput);`

Reply:
Resuelto